crates/ty_python_semantic/src/program.rs

```diff
@@ -23,23 +27,34 @@ pub struct Program {
 impl Program {
     pub fn from_settings(db: &dyn Db, settings: ProgramSettings) -> anyhow::Result<Self> {
         let ProgramSettings {
-            python_version,
+            python_version: python_version_with_source,
             python_platform,
             search_paths,
         } = settings;
 
-        tracing::info!("Python version: Python {python_version}, platform: {python_platform}");
+        tracing::info!(
+            "Python version: Python {python_version}, platform: {python_platform}",
+            python_version = python_version_with_source.version
+        );
 
         let search_paths = SearchPaths::from_settings(db, &search_paths)
             .with_context(|| "Invalid search path settings")?;
 
         Ok(
-            Program::builder(python_version, python_platform, search_paths)
+            Program::builder(python_version_with_source, python_platform, search_paths)
                 .durability(Durability::HIGH)
                 .new(db),
         )
     }
 
+    pub fn python_version(self, db: &dyn Db) -> PythonVersion {
+        self.python_version_with_source(db).version
+    }
+
+    pub fn python_version_source(self, db: &dyn Db) -> &ValueSource {
+        &self.python_version_with_source(db).source
+    }
```

Comment:
Nit: I would delete this method and directly use `python_version_with_source` (given that it is public too). It's only used in one place anyway which calls both `python_version` and `python_version_with_source`